config/config-mysql_apache_jvm.yaml

```diff
@@ -0,0 +1,307 @@
+# This config mimicks the Cloud Monitoring Agent by collecting metrics from popular third-party services the Agent already supports
+# and formatting the metrics in the same way, so they appear similarly in Cloud Monitoring. The services targeted here are MySQL, Apache and JVM.
+# There is also support for DogStatsD metrics, with no transformations
+# Please remove/comment-out the pipelines you don't need/want in the last section.
+# Documentation specific to each exporter is found in the comment directly above its configuration
+
+receivers:
+
+  hostmetrics:
+    collection_interval: 60s
+    scrapers:
+      cpu:
+      filesystem:
+      load:
+      memory:
+      network:
+      swap:
+
+  # https://github.com/prometheus/mysqld_exporter
+  prometheus_exec/mysql:
+  # --no-collect.global_variables reduces the amount of generated metrics, can be removed to enable all (600+) MySQL metrics
+    exec: ./prometheus_exporters/mysqld_exporter --web.listen-address=":{{port}}" --no-collect.global_variables
+    port: 9104
+    env:
+      # DATA_SOURCE_NAME and its value are needed to connect to the MySQL server, alternatively you could use a .my.cnf configuration file
```

Comment:
I don't have any context but is it standard to use a .my.cnf file? If so should we leave this commented out by default?

Reply:
The exporter indicates to use either: https://github.com/prometheus/mysqld_exporter

The only extra config you need for the exporter to work is the `DATA_SOURCE_NAME`, so using it through an env variable I think is more convenient! I just let the user know they can use a config file if they'd like.

Reply:
Yea I didn't give very good context there but the main reason I asked is because its better if everything can work by default (pending env var setup). If possible it'd be great if there's nothing in the config file that a user NEEDS to change to get it workign

Reply:
You're right! Technically if the user has all their 3rd-party applications on default values and already has a `.my.cnf` file or the `DATA_SOURCE_NAME` environment variable, it could be run without opening the config file. I think it's highly unlikely, given there is also some JMX setup needed and one or more pipelines might be removed. I'll comment them out for now, since the user will probably read the file anyways, but the off-chance they run it knowing their `env` vars are already setup, my dummy `env` var would mess it up. So yeah, I'll comment it out!